src/lib/list/selection-list.ts

```diff
@@ -274,6 +324,62 @@ export class MatSelectionList extends _MatSelectionListMixinBase
     }
   }
 
+  /** Reports a value change to the ControlValueAccessor */
+  _reportValueChange() {
+    if (this.options) {
+      this._onChange(this._getSelectedOptionValues());
+    }
+  }
+
+  /** Emits a change event if the selected state of an option changed. */
+  _emitChangeEvent(option: MatListOption) {
+    this.change.emit(new MatSelectionListChange(this, option));
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  writeValue(values: string[]): void {
+    if (this.options) {
+      this._setOptionsFromValues(values || []);
+    }
+  }
+
+  /** Implemented as a part of ControlValueAccessor. */
+  setDisabledState(isDisabled: boolean): void {
+    if (this.options) {
+      this.options.forEach(option => option.disabled = isDisabled);
+    }
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnChange(fn: any): void {
+    this._onChange = fn;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  /** Returns the option with the specified value. */
+  private _getOptionFromValue(value: string): MatListOption | undefined {
+    return this.options.find(option => option.value === value);
+  }
+
+  /** Sets the selected options based on the specified values. */
+  private _setOptionsFromValues(values: string[]) {
+    this.options.forEach(option => option._setSelected(false));
+
+    values
+      .map(value => this._getOptionFromValue(value))
+      .filter(option => !!option)
+      .forEach(option => option!._setSelected(true));
+  }
+
+  /** Returns the values of the selected options. */
+  private _getSelectedOptionValues(): string[] {
```

Comment:
Consider inlining this since it's only used in one place and it's a one-liner?

Reply:
I'd personally prefer keeping this as a function. It makes it the code more sorted with the `_setOptionsFromValues` method.